src/args/argbuilder/flag.rs

```diff
@@ -24,15 +24,101 @@ pub struct FlagBuilder<'n> {
     /// The short version (i.e. single character)
     /// of the argument, no preceding `-`
     pub short: Option<char>,
-    pub global: bool,
     /// A list of names for other arguments that *mutually override* this flag
     pub overrides: Option<Vec<&'n str>>,
-    pub hidden: bool
+    pub settings: ArgFlags,
+}
+
+impl<'n> FlagBuilder<'n> {
+    pub fn new(name: &'n str) -> Self {
+        FlagBuilder {
+            name: name,
+            short: None,
+            long: None,
+            help: None,
+            blacklist: None,
+            requires: None,
+            overrides: None,
+            settings: ArgFlags::new()
+        }
+    }
+}
+
+impl<'n, 'a> From<&'a Arg<'n, 'n, 'n, 'n, 'n, 'n>> for FlagBuilder<'n> {
+    fn from(a: &Arg<'n, 'n, 'n, 'n, 'n, 'n>) -> Self {
+        if a.validator.is_some() {
+            panic!("The argument '{}' has a validator set, yet was parsed as a flag. Ensure \
+                    .takes_value(true) or .index(u8) is set.")
+        }
+        if !a.empty_vals {
+            // Empty vals defaults to true, so if it's false it was manually set
+            panic!("The argument '{}' cannot have empty_values() set because it is a flag. \
+                Perhaps you mean't to set takes_value(true) as well?", a.name);
+        }
+        if a.required {
+            panic!("The argument '{}' cannot be required(true) because it has no index() or \
+                takes_value(true)", a.name);
+        }
+        if a.possible_vals.is_some() {
+            panic!("The argument '{}' cannot have a specific value set because it doesn't \
+            have takes_value(true) set", a.name);
```

Comment:
Can these lines have consistent formatting? One has the next line starting from the previous line's quotation mark, two of them are indented, one is on the same line.